components/nimbus/src/metrics.rs

```diff
@@ -0,0 +1,76 @@
+use crate::{enrollment::ExperimentEnrollment, EnrollmentStatus};
+use serde_derive::{Deserialize, Serialize};
+
+pub trait MetricsHandler: Send + Sync {
+    fn record_enrollment_statuses(&self, enrollment_status_extras: Vec<EnrollmentStatusExtraDef>);
+}
```

Comment:
Is there a plan to use `MetricsHandler` in the `stateless` API too? Perhaps we should work towards that future.

Reply:
Yes that is the plan, I'm just planning to handle it in a separate PR/ticket since that part goes across repos https://mozilla-hub.atlassian.net/browse/EXP-3828